sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -337,20 +395,370 @@ private void CloseLink(ReceivingAmqpLink link)
             }
         }
 
+        /// <summary>Indicates that the receiver wants to defer the processing for the message.</summary>
+        ///
+        /// <param name="message">The lock token of the <see cref="ServiceBusMessage" />.</param>
+        /// <param name="propertiesToModify">The properties of the message to modify while deferring the message.</param>
+        /// <param name="cancellationToken"></param>
+        ///
+        /// <remarks>
+        /// A lock token can be found in <see cref="ServiceBusReceivedMessage.LockToken"/>,
+        /// only when <see cref="ReceiveMode"/> is set to <see cref="ReceiveMode.PeekLock"/>.
+        /// In order to receive this message again in the future, you will need to save the <see cref="ServiceBusReceivedMessage.SequenceNumber"/>
+        /// and receive it using <see cref="ServiceBusReceiver.ReceiveDeferredMessageAsync(long, CancellationToken)"/>.
+        /// Deferring messages does not impact message's expiration, meaning that deferred messages can still expire.
+        /// This operation can only be performed on messages that were received by this receiver.
+        /// </remarks>
+        public override async Task DeferAsync(
+            ServiceBusReceivedMessage message,
+            IDictionary<string, object> propertiesToModify = null,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(ServiceBusReceiver));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
```

Comment:
Is anything other than `TaskCanceledException` that will be thrown by this extension method?